src/Carbonara/Services/CalculationService/CalculationService.cs

```diff
@@ -93,13 +98,16 @@ public async Task<TotalCalculationResult> CalculateTotalSummary(List<string> txH
                 return result;
             }
 
-            var taskResults = new List<Task<TotalCalculationResult>>();
+            var waitingTime = int.Parse(_configuration["RequestWaitingTime"]);
+
+            var results = new List<TotalCalculationResult>();
             foreach (var txHash in txHashes)
             {
-                taskResults.Add(Calculate(txHash, hashingAlg, countryToUseForCo2EmissionAverage));
-            }
+                var calculationResult = await Calculate(txHash, hashingAlg, countryToUseForCo2EmissionAverage);
+                results.Add(calculationResult);
 
-            var results = (await Task.WhenAll(taskResults)).ToList();
+                Thread.Sleep(waitingTime);

```

Comment:
please leave a comment as to why we take this approach because the code looks really stupid, but there is a reason for that.